### PR TITLE
feat: add aws role to eks cluster auth

### PR DESCRIPTION
**Pull request by Automaat**

### CODE REVIEW

File: .github/workflows/perf-test-eks.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
       - name: make dev/tools
         run: make dev/tools
       - name: Start cluster
-        run: ENV=eks make start-cluster
+        run: ENV=eks TF_VARS="aws_role_id=${{ secrets.AWS_ACCOUNT_ID }}" make start-cluster
```
Comment (michaelbeaumont) [inline .github/workflows/perf-test-eks.yaml:34]: How is the account id a role id?

Comment (Automaat) [reply .github/workflows/perf-test-eks.yaml]: secret naming is actually wrong, it holds ROLE id atm, changed

File: infrastructure/eks/main.tf [hunk 2]
```diff
@@ -55,6 +55,16 @@ module "eks" {
       desired_size = var.nodes_number
     }
   }
+
+  manage_aws_auth_configmap = true
+
+  aws_auth_roles = [
```
Comment (michaelbeaumont) [inline infrastructure/eks/main.tf:61]: Isn't the creator always added to `system:masters` for the EKS cluster?

File: .github/workflows/perf-test-eks.yaml [hunk 3]
```diff
@@ -20,7 +20,7 @@ jobs:
       - name: Configure AWS Credentials
         uses: aws-actions/configure-aws-credentials@v2
         with:
-          role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_ID }}:role/mesh-perf-ci
+          role-to-assume: arn:aws:iam::${{ secrets.AWS_ROLE_ID }}:role/mesh-perf-ci
```
Comment (michaelbeaumont) [inline .github/workflows/perf-test-eks.yaml:23]: but wait, an account id belongs here I think

File: infrastructure/eks/main.tf [hunk 4]
```diff
@@ -55,6 +55,16 @@ module "eks" {
...
+    {
+      rolearn  = "arn:aws:iam::${ var.aws_role_id }:role/mesh-perf-ci"
```
Comment (michaelbeaumont) [inline infrastructure/eks/main.tf:63]: I think you can avoid passing in an account id with something like

```
data "aws_caller_identity" "current" {}
locals {
    account_id = data.aws_caller_identity.current.account_id
}
```

**Pull request closed without merging**